src/mappings/trades.ts

```diff
@@ -58,27 +58,6 @@ export function onTrade(event: TradeEvent): void {
   createSolutionOrAddTrade(trade, event)
 }
 
-export function onTradeReversion(event: TradeReversionEvent): void {
-  let orderId = toOrderId(event.params.owner, event.params.orderId)
-  log.info('[onTradeReversion] New Trade Reversion for orderId {}', [orderId])
-
-  // TODO: Handle trade revertion, not working
-  // Subgraph instance failed to run: Failed to process trigger in transaction 0x86a4…2f6b: Failed to handle Ethereum event with handler "onTradeReversion": Mapping aborted at ~lib/@graphprotocol/graph-ts/index.ts, line 1046, column 4, with message: Value is not an array., code: SubgraphSyncingFailure, id: QmTPBbCq2WhFFP54XVWHPiwosNrdtxiyyojNtDNMD9jRaD
-
-  // // Get the trade that need to be reverted (only one that is active in the batch)
-  // let batchId = getBatchId(event)
-  // let trade = getActiveTradeInBatch(orderId, batchId)
-  // if (trade == null) {
-  //   // If the contract emit a revert it should be
-  //   throw new Error('The order ' + orderId + " doesn't have any active trade to revert")
-  // }
-
-  // // Set the revert date for the active order
-  // trade.revertEpoch = event.block.timestamp
-  // trade.save()
-  // log.info('[onTradeReversion] Reverted trade {}', [trade.id])
-}
-
 function _createTrade(orderId: string, event: TradeEvent): Trade {
   let params = event.params
```

Comment:
I would really like to see trades have a `solution` field. But I like your suggestion of keeping the solution's `trades` field as is (and **not** a `derivedFrom` field) as we use it as an append only list (and is just as easy to keep in sync as a potential trade count field).

Reply:
https://github.com/gnosis/dex-subgraph/issues/126